### DISCUSSION TITLE
JK-BMS Negative Temperature Bug

### DISCUSSION THREAD

**Discussion by karikolehmainen** [Louisvdw/dbus-serialbattery#1156] [General]
Hi, 

I am witnessing very odd behaviour with JK-BMS B2a24s20p connected to Cerbo via Bluetooth. First of all I think I may have bad cell on my 16s batterypack as one cell is systematically either lowest or highest voltage when under load or high charge current (105Ah EVE cells). That may very well be an cell issue but the more odd thing is that I see cell temperature as 100°C when cells go bellow 0°C. Battery is deployed off-grid in wilderness and I do not have chance to see it with Bluetooth App if the same values are shown with that. Has anyone seen similar behaviour? There are two possibilities, either bug in JK-BMS firmware or bug in serialbattery driver on Cerbo.
<img width="732" alt="Screenshot 2024-12-20 at 9 38 23" src="https://github.com/user-attachments/assets/cac98b7c-889e-47e0-a38c-9c212a82d22f" />
Note the 50°C step happens when the other of the two temperature sensors flips to negative value and starts reading the 100°C bogus value. It seems that either the firmware or driver averages the output of the two sensors. I do not like that logic, It can hide potential problems in the battery and reduce reliability. It would be then better to have just one sensor rather than two.

**Comment by RichGorrod** [comment]
Hi.
I think I remember seeing Andy @ off grid garage have this problem with a bad temperature sensor. Will try and find the link.

Rich
